src/cursor/cur_hs.c

```diff
@@ -883,6 +883,12 @@ __curhs_insert(WT_CURSOR *cursor)
         goto retry;
     WT_ERR(ret);
 
+    /*
```

Comment:
Other than diagnostic builds, do we expect a failure that can happen after insertion? If not, can we move all this new code also into the diagnostic code flow?

Reply:
`reset` returns 0 so it can happen there since we're wrapping with `WT_TRET`.

Reply:
Thanks for the details. Understood.